File: src/main/java/com/github/ianiiaannn/operationalrelife/handlers/ModEventHandler.java
package com.github.ianiiaannn.operationalrelife.handlers;


import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;


public class ModEventHandler {
  private static final IEventBus BUS = FMLJavaModLoadingContext.get().getModEventBus();
  public static void registerAll(){
    ModEventHandler.BUS.register(ModEventHandler.class);
    ModEventHandler.BUS.register(new ModEventHandler());
  }
  @SubscribeEvent
  public  static  void onCommonSetup(FMLCommonSetupEvent event){
    ForgeEventHandler.registerEvents();
  }

}

File: src/main/kotlin/dev/iancmd/operationrelife/handlers/ModEventHandler.kt
package dev.iancmd.operationrelife.handlers

import net.minecraftforge.eventbus.api.IEventBus
import net.minecraftforge.eventbus.api.SubscribeEvent
import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext

class ModEventHandler {


  @SubscribeEvent
  @Suppress("UNUSED_PARAMETER")
  fun onCommonSetup(event: FMLCommonSetupEvent?) {
    ForgeEventHandler.registerEvents()
  }
  companion object {
    private val BUS: IEventBus = FMLJavaModLoadingContext.get().modEventBus
    fun registerAll() {
      BUS.register(ModEventHandler::class.java)
    }
  }
}
